sympy/plotting/series.py

```diff
@@ -73,25 +132,803 @@ class BaseSeries:
     is_generic = False
     # Represent generic user-provided numerical data
 
-    def __init__(self):
-        super().__init__()
+    is_vector = False
+    is_2Dvector = False
+    is_3Dvector = False
+    is_slice = False
+    # Represents a 2D or 3D vector data series
```

Comment:
I don't think that these are scalable patterns, because adding every new subclasses need to set this in the base class. I'd try to avoid this by `isinstance` check and inheritance, if the object oriented programming is used anyway.

Reply:
There are many vector classes, which can be classified as 2D_vector or 3D_vector. Base class is using these attributes in order to not use `isinstance`. It makes the code cleaner and easier to read.

Reply:
I also recognize that the property isn't used in any place

Reply:
`is_3Dvector` is used into `series.py`. The others will be used on other files that will be added in future PRs.